tests/unit/s3/test_key.py

```diff
@@ -150,6 +154,72 @@ def test_change_storage_class_new_bucket(self):
             preserve_acl=True,
             validate_dst_bucket=True,
         )
+    
+    def test_download_succeeds(self):
+        test_case_headers = [
+            [('Content-Length', '5')],
+            [('Content-Range', 'bytes 15-20/100')],
+        ]
+        for headers in test_case_headers:
+            with self.subTest(headers=headers):
+                head_object_response = self.create_response(
+                    status_code=200, header=headers)
+
+                media_response = self.create_response(
+                    status_code=200, header=headers)
+                media_response.read.side_effect = [
+                    b'12345',
+                    b'',
+                    b'',  # Key.close call.
+                ]
+
+                self.https_connection.getresponse.side_effect = [
+                    head_object_response,
+                    media_response
+                ]
+
+                bucket = Bucket(self.service_connection, 'bucket')
+                key = bucket.get_key('object')
+
+                output_stream = io.BytesIO()
+                key.get_file(output_stream)
+
+                output_stream.seek(0)
+                self.assertEqual(output_stream.read(), b'12345')
+    
+    def test_download_raises_retriable_error_with_truncated_stream(self):
```

Comment:
Can you please leave a comment on how this test sets up a truncated stream. Is it because of just 3 bytes present instead of 5?

Reply:
That's right! Left a comment.